server/Episememe.Application/Features/FileUpload/FileUploadCommandHandler.cs

```diff
@@ -91,6 +87,25 @@ private async Task CreateMediaInstanceInDatabase(string id, string dataType, IEn
             await _context.SaveChangesAsync(CancellationToken.None);
         }
 
+        private string GenerateMediaInstanceId()
+        {
+            var idExists = true;
+            var newId = string.Empty;
+            var existingIds = _context.MediaInstances.Select(mi => mi.Id);
+
+            while (idExists)
```

Comment:
This entire "generate ids multiple times" was not meant to resolve the problem of database duplicates. It was meant to help e.g. if 2 threads generate the exact same ID, and one of them creates a file. The other thread will fail, and this loop won't help.

Generate an ID. Create a new file for that ID. If it fails, go back and pick a new ID